src/compiler/checker.ts

```diff
@@ -11191,7 +11194,7 @@ namespace ts {
         }
 
         function createInferenceContext(signature: Signature, flags: InferenceFlags, compareTypes?: TypeComparer, baseInferences?: InferenceInfo[]): InferenceContext {
-            const inferences = baseInferences ? map(baseInferences, cloneInferenceInfo) : map(signature.typeParameters, createInferenceInfo);
+            const inferences = baseInferences ? baseInferences.map(cloneInferenceInfo) : signature.typeParameters.map(createInferenceInfo);

```

Comment:
Looks like this function should never be called if `signature.typeParameters` is undefined, since we expect `inferences` to always be defined elsewhere.